packages/vite-plugin-svelte/src/plugins/setup-optimizer.js

```diff
@@ -32,7 +32,8 @@ export function setupOptimizer(api) {
 	return {
 		name: 'vite-plugin-svelte:setup-optimizer',
 		apply: 'serve',
-		config() {
+		configEnvironment(_name, config) {
+			const consumer = config.consumer ?? 'client';
```

Comment:
`configEnvironment` doesn't apply the default consumer here https://github.com/vitejs/vite/blob/557f797d29422027e8c451ca50dd84bf8c41b5f0/packages/vite/src/node/config.ts#L787-L789 so this always evaluates to "client" for `@cloudflare/vite-plugin` since it doesn't explicitly set the `consumer` option but instead relies on Vite setting it to the default of "server" when the options are fully resolved (after this hook?).

Reply:
great catch, i guess we'll have to replicate the code from vite.